src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/EmbeddingProviderFactory.java

```diff
@@ -117,10 +118,26 @@ private synchronized EmbeddingProvider addService(
       throw ErrorCodeV1.VECTORIZE_SERVICE_TYPE_UNAVAILABLE.toApiException(
           "unknown service provider '%s'", configuration.serviceProvider());
     }
+
+    // Get the provider, then get the model.
+    var providerConfig = embeddingProvidersConfig.providers().get(configuration.serviceProvider());
+    if (providerConfig == null) {
+      throw ErrorCodeV1.VECTORIZE_SERVICE_TYPE_UNAVAILABLE.toApiException(
+          "unknown service provider '%s'", configuration.serviceProvider());
+    }
+    EmbeddingProvidersConfig.EmbeddingProviderConfig.ModelConfig model =
+        embeddingProvidersConfig.providers().get(configuration.serviceProvider()).models().stream()
```

Comment:
ah, this is ugly. The ProviderConfig does not have the name on it, I'll fix this later.